[Lin Shuang Shuang] iP

Lin-Shuang-Shuang opened the pull request:
## LeryChatbotPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

LeryChatbotPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 - [x] Managing todo tasks
 - [ ] Managing deadlines
 - [ ] Managing events
 - [ ] Reminders (coming soon)

***

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Lin-Shuang-Shuang committed:
Level 0. Rename, Greet, Exit

Lin-Shuang-Shuang committed:
Level 1. Echo

Lin-Shuang-Shuang committed:
Level 2. Add, List

Lin-Shuang-Shuang committed:
Level 3. Mark as Done

Lin-Shuang-Shuang committed:
Level 4. ToDos, Events, Deadlines

Lin-Shuang-Shuang committed:
Automated Text UI Testing

Lin-Shuang-Shuang committed:
Level 5. Handle Errors

Lin-Shuang-Shuang committed:
Level 6. Delete

Lin-Shuang-Shuang committed:
Add saving tasks in and loading tasks from hard disk feature

Lin-Shuang-Shuang committed:
Merge branch 'branch-Level-7'

Lin-Shuang-Shuang committed:
Add dates and times to deadline tasks

Lin-Shuang-Shuang committed:
Merge branch 'branch-Level-8'

Lin-Shuang-Shuang committed:
Add more OOP

Lin-Shuang-Shuang committed:
Merge branch 'branch-AMoreOOP'

Lin-Shuang-Shuang committed:
Add packages

Lin-Shuang-Shuang committed:
Merge branch 'branch-APackages'

Lin-Shuang-Shuang committed:
Merge branch 'add-gradle-support'

Lin-Shuang-Shuang committed:
Add Gradle support

Lin-Shuang-Shuang committed:
Add JUnit test cases for task

Lin-Shuang-Shuang committed:
Merge branch 'branch-AJUnit'

Lin-Shuang-Shuang committed:
Package App as a JAR file

Lin-Shuang-Shuang committed:
Add Javadoc comments

Lin-Shuang-Shuang committed:
Modified code to comply with Java coding standard

Lin-Shuang-Shuang committed:
Add find function

Lin-Shuang-Shuang committed:
Merge branch 'branch-A-JavaDoc'

Lin-Shuang-Shuang committed:
Solve merge conflicts

Lin-Shuang-Shuang committed:
Merge branch 'branch-Level-9'

Lin-Shuang-Shuang committed:
Use checkstyle

Lin-Shuang-Shuang committed:
Merge branch 'branch-A-CheckStyle'

jieqiboh commented on src/main/java/lery/task/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+        String list = "Here are the tasks in your list:\n";
+        for (int i = 0; i < this.getSize(); i++) {
+            Task task = this.getTask(i);
+            String message = Integer.toString(i + 1) + "." + "[" + task.getType()
```
Perhaps the Integer.toString call is not necessary here, since i is already an integer 😄

jieqiboh commented on src/main/java/lery/task/Task.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @return a message indicating the status change.
+     */
+    public String markAsDone() {
```
I think it might be more suitable to handle string output in your parser 😃

jieqiboh commented on src/main/java/lery/task/Todo.java:
```diff
@@ -0,0 +1,56 @@
+     * @return an empty string.
+     */
+    @Override
+    public String getExtraInfoShortened() {
```
Perhaps extra information is a bit vague, and could use some elaboration on what information it is referring to? 😄

jieqiboh commented on src/main/java/lery/task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+ */
+public class Deadline extends Task {
+    private static final String TYPE = "D";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
I like your use of regex here! 😄

jieqiboh commented on src/main/java/lery/Ui.java:
```diff
@@ -0,0 +1,46 @@
+ *
+ */
+public class Ui {
+    private static final String LINE = "____________________________________________________________\n";
```
```suggestion
    private static final String BORDER = "_".repeat(50);
```

I think it would help to make this into a constant with fixed width like so

jieqiboh commented on src/main/java/lery/Parser.java:
```diff
@@ -0,0 +1,158 @@
+    public String parseCommand(String command) throws LeryException {
+
+        TaskList taskList = this.storage.getTaskList();
+        if (command.equalsIgnoreCase("list")) {
```
Just a suggestion, but I think using a switch statement would look neater 😃

jieqiboh commented on src/main/java/lery/Storage.java:
```diff
@@ -0,0 +1,150 @@
+     *
+     * @param t the task to be deleted.
+     */
+    public void delete(Task t) {
```
Perhaps deleteTask would be more apt, following your previous naming conventions 😄 
```suggestion
    public void deleteTask(Task t) {
```

jieqiboh reviewed (commented):
Overall, there were minimal coding standard violations, and everything was well formatted, with JavaDocs being used widely 😄 I feel that some methods could use more detailed naming, such as `getExtraInfo`, and some things like outputting strings could be handled by the Ui instead.

marcus-ny commented on src/main/java/lery/Parser.java:
```diff
@@ -0,0 +1,158 @@
+            } catch (IndexOutOfBoundsException e) {
+                throw new LeryException("Erm... Please enter a task number.");
+            }
+
```
Just a nitpick but there are similar empty lines in some parts that are not consistent. Perhaps you could configure an IDE formatter to do this for you automatically on save?

marcus-ny commented on src/main/java/lery/Parser.java:
```diff
@@ -0,0 +1,158 @@
+
+        msg = msg + "[" + newTask.getType() + "]" + "[" + newTask.getStatusIcon()
+                + "]" + " " + newTask.getDescription() + newTask.getExtraInfo()
+                + "\nNow you have " + Integer.toString(this.storage.getSize())
```
It's great that you used line-breaks to make long expressions like this more readable 👍

marcus-ny commented on src/main/java/lery/task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    /**
+     * Gets the extra information of the deadline task.
```
Perhaps the header comments here could be more descriptive to explain what this extra information is for this particular subclass

marcus-ny commented on src/main/java/lery/task/Event.java:
```diff
@@ -0,0 +1,74 @@
+ * Represents an event task in the Lery chatbot application,
+ * which is a specific type of task associated with a start and end time.
+ *
+ * It extends the generic Task class and provides methods to get the task type,
```
I like that the header comments that explain the functionality and expected behaviour of the class in detail 👍

marcus-ny reviewed (commented):
Overall, close to no violations of coding standard at all! 👍 Just a few nitpicks with the line spaces but these can be formatted in your IDE automatically.

Lin-Shuang-Shuang committed:
Create GUi for Lery

Lin-Shuang-Shuang committed:
Merge branch 'branch-Level-10'

Lin-Shuang-Shuang committed:
Add some assert statements to code

Use assert to
- validate assumptions that should hold at various points in the code
- enhance debugging capabilities
- improve code maintainability and catch potential issues early on

Lin-Shuang-Shuang committed:
Improve code quality

If-else blocks are used heavily throughout the loadTasks method in the
Storage class. The same statement that checks the content of each line
in the text file is used repeatedly at every else-if block.

Let's clean up the code by replacing the if-else statements with a
switch statement so that the code looks more tidy and readable.

This reduces repetition as well. Making the code look more readable and clean allows other people to
better understand it and improves the maintainability of the code.

Lin-Shuang-Shuang committed:
Merge pull request #1 from Lin-Shuang-Shuang/branch-A-Assertions

Add some assert statements to code

Lin-Shuang-Shuang committed:
Merge branch 'master' into branch-A-CodeQuality

Lin-Shuang-Shuang committed:
Merge pull request #2 from Lin-Shuang-Shuang/branch-A-CodeQuality

Improve code quality

Lin-Shuang-Shuang committed:
Set up CI

Lin-Shuang-Shuang committed:
Sort task list

Task list is unsorted and the order depends on the time the task was
added. User may find it difficult to locate tasks if list is long.

Let's add a command for sort that sorts tasks by their task type and
deadline. If task type is deadline, they will be located at the start of
the list and sorted by their deadline starting from the earliest
deadline to latest.

This allows user to locate tasks easily, and can be reminded of the
tasks with the earliest deadline so that they can complete them on time.

Lin-Shuang-Shuang committed:
Merge branch 'branch-C-Sort'

Lin-Shuang-Shuang committed:
Improve GUI

Lin-Shuang-Shuang committed:
Merge branch 'branch-A-BetterGui'

Lin-Shuang-Shuang committed:
Give Lery a personality

Lin-Shuang-Shuang committed:
Merge branch 'branch-A-Personality'

Lin-Shuang-Shuang committed:
Replace Duke with Lery

Lin-Shuang-Shuang committed:
Add product name to the title bar

Lin-Shuang-Shuang committed:
Resolve checkstyle error

Lin-Shuang-Shuang committed:
Add user guide

Lin-Shuang-Shuang committed:
Merge branch 'branch-A-UserGuide'

Lin-Shuang-Shuang committed:
Fix Storage functions

Lin-Shuang-Shuang committed:
Fix text file accessibility error

Lin-Shuang-Shuang committed:
Release product on GitHub
